tests/test_forms.py

```diff
@@ -759,9 +758,9 @@ async def extract_my_slot(
         (["my_slot", "other_slot"], {}, [SlotSet(REQUESTED_SLOT, "other_slot")]),
         # Extract method for slot which is also mapped in domain
         (
-            ["my_slot"],
+            ["my_slot", "other_slot"],
             {"forms": {"some_form": {"required_slots": ["my_slot"]}}},
-            [],
+            [SlotSet(REQUESTED_SLOT, "other_slot")],
```

Comment:
How come you needed to change these lines?

Reply:
Unfortunately, this PR contains two changes:
1. Support async tests
2. My task-related changes.

By enabling `asyncio_mode`, this test started to fail, and I needed to fix it.

Reply:
Is this fixing of the test still in line with the original test or did the fix change what is being tested here?

Reply:
Good point. 
Without my changes test fails because this part doesn't trigger.
```
    with pytest.warns(UserWarning):
        events = await form.run(dispatcher=dispatcher, tracker=tracker, domain=domain)
```

Reply:
I did a brief check and it seems to me that the test is already failing on the spaces branch 🤔, but it doesn't fail on main. Can you confirm this? I think by changing this test here, we might just be covering up an issue.

Reply:
`main` failed for me, with the same error.
Can you try again to run on the `main` branch, please update `pyproject.toml` with 
```
 [tool.pytest.ini_options]
 asyncio_mode = "auto"
```

Reply:
Fully agree with you